### PR TITLE
Added support for non-AWS hosted domains

### PR DESCRIPTION
**Pull request by basschipper**
Like your SES provider!

But, I try to get this working for domain names that are not hosted by AWS. For that I made the following proposal, what do you think?

### CODE REVIEW

**Comment by basschipper** [comment]
Hi Mark,

Thank you for the feedback!

This PR supports domain names that are not hosted by AWS (Route53). Instead of a reference to the `HostedZoneId` you specify the `DomainName` directly and no DKIM records are created in Route53. The DKIM records will then only be generated and then have to be entered manually at your own DNS provider.

It seemed more logical to look at SES from the `Domain` perspective rather than from the 'DKIM' perspective. Hence the changes to the resource name. However, this (`Custom::DKIM`) still works because it simply falls into the else of the `ResourceType` condition in `src/ses.py`. However, I have not fully tested the `Custom::DKIM`.

Lastly, support for identity policies have been added, so that you can add policies in SES that enable cross-account mailing.


```yaml
  SesDomain:
    Type: Custom::Domain
    Properties:
      DomainName: foobar.nl
      DKIM: True
      ServiceToken: !Sub 'arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:cfn-ses-provider'
  SesPolicy:
    Type: Custom::IdentityPolicy
    Properties:
      Identity: !Ref SesDomain
      PolicyName: cross-account-allow
      PolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Principal:
              AWS:
                - 'arn:aws:iam::1234:root'
                - 'arn:aws:iam::4321:root'
            Action:
              - ses:SendEmail
              - ses:SendRawEmail
            Resource: !Sub 'arn:aws:ses:${AWS::Region}:${AWS::AccountId}:identity/${SesDomain}'
      ServiceToken: !Sub 'arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:cfn-ses-provider'
```

File: src/cfn_identity_policy_provider.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    def __init__(self):
+        self.Effect = None
+        self.Principal = None
+        self.Action = None
+        self.Resource = None
+
+    @classmethod
+    def from_dict(cls, dict):
+        policy = cls()
+        policy.Effect = dict['Effect'] if 'Effect' in dict else None
```
Comment (mvanholsteijn) [inline src/cfn_identity_policy_provider.py]: can be written as: dict.get('Effect')

Comment (basschipper) [reply src/cfn_identity_policy_provider.py]: Fixed. Much cleaner, thanx!

**Comment by mvanholsteijn** [comment]
Hi @basschipper, my apologies for not merging your changes more than 1 year ago, Since v0.6.4 I have modified the provider to deprecate the Custom::DKIM  in favor of the Custom::DomainIdentity.

the original DKIM resource was one of the first I ever created and it violates the principle of CloudFormation that a custom::resource should manage exactly one  resource.  It becomes brittle when a custom resource manages multiple resources as Custom::DKIM did. 

Anyway, thank you for using the provider and putting in the work for this PR.

Cheers,

Mark

**Pull request closed without merging**